ivy/stateful/optimizers.py

```diff
@@ -476,7 +484,7 @@ def __init__(
 
```

Comment:
Since device follows fallback_to_non_compiled, it should be passed as a keyword argument (device=device).